src/Chapter3.hs

```diff
@@ -970,6 +1114,27 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data Weekday
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday 
+  deriving (Show, Eq, Bounded, Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend Saturday  = True
+isWeekend Sunday    = True
+isWeekend _         = False
+
+nextDay :: Weekday -> Weekday
+nextDay weekday = if weekday == maxBound then minBound else succ weekday 
```

Comment:
Thank you for suggestion. I read the document on minBound and maxBound and changed the code to use them.